### PR TITLE
feat: DOMA-6389 organization types added

### PR DESCRIPTION
**Pull request by SavelevMatthew**
## What's inside?
- Separated all organisations into 3 types: `MANAGING_COMPANY`, `HOLDING` and `SERVICE_PROVIDER`.
- Added optional type prop to `registerNewOrganization` mutation
- Fixed resident connection and property map resolve bugs.
- Partly refactored tests in Organization / Property domains.
- Removed legacy `CheckIfPropertyAddressExistsService`

## About organization types:
- `MANAGING_COMPANY` is default organization type, which live in v1. Nothing new here.
- `HOLDING` is organization type for parent organizations in `OrganizationLink`. No extra logic will be added. Added to restrict organization-contractor flow, which will occur later (need to restrict holdings to invite contractors in parent organization because of architectural complexity)
- `SERVICE_PROVIDER` is type for contractors / settlement centers / resource providers. They will live on v1 as well, but will see slightly different interface.

## About residents.
- I've tried to change as little as possible to make everything work together, so I expect some of domains to not work properly with `SERVICE_PROVIDER `. (They'll be modified later "per domain" as `SERVICE_PROVIDER` part of v1 will grow_
- However it's not the case for resident domain, since we're having a mobile app, which must be working constantly.
- For now residents has no conflicts with orgazanition types, since CRM part of `SERVICE_PROVIDER` is empty.
- The only requirement is that **all residents must be connected / reconnected to `MANAGING_COMPANY` properties only**, since we're focusing on them right now.

## TODO

- [x] Add 3 organization types: `HOLDING`, `MANAGING_COMPANY` and `SERVICE_PROVIDER`
- [x] Migrate existing organizations manually (set `HOLDING` for each org existing with `OrganizationLink`)
- [x] Make sure existing tests are passing
- [x] Cover organization domain with new tests
- [x] Cover resident/property domain with tests
- [x] Make sure that API is backward compatible (`MANAGING_COMPANY` is created by default)
- [x] Make sure residents connects only to properties of `MANAGING_COMPANY`

### CODE REVIEW

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/organization/schema/OrganizationLink.js [hunk 1]
```diff
@@ -50,6 +51,16 @@ const OrganizationLink = new GQLListSchema('OrganizationLink', {
         delete: false,
         auth: true,
     },
+    hooks: {
+        validateInput: async ({ resolvedData, addValidationError }) => {
+            if (resolvedData['from']) {
+                const org = await getById('Organization', resolvedData['from'])
+                if (!org || org.type !== HOLDING_TYPE) {
+                    addValidationError(`Parent organization ("from") must have "${HOLDING_TYPE}" type`)
```
Comment (Alllex202) [inline apps/condo/domains/organization/schema/OrganizationLink.js:59]: you should use `GQLError`

File: apps/condo/domains/organization/schema/Organization.js [hunk 2]
```diff
@@ -36,6 +37,20 @@ const Organization = new GQLListSchema('Organization', {
             isRequired: true,
             kmigratorOptions: { null: false },
         },
+        type: {
+            schemaDoc: 'Type of organization',
```
Comment (pahaz) [inline apps/condo/domains/organization/schema/Organization.js:41]: It's better to write more info about this tricky field )

File: apps/condo/domains/organization/schema/OrganizationLink.js (hunk 1 above)
Comment (pahaz) [reply apps/condo/domains/organization/schema/OrganizationLink.js:59]: It's better to use our own GQL messages to help others developers.

File: apps/condo/domains/organization/schema/Organization.js [hunk 3]
```diff
@@ -36,6 +37,20 @@ const Organization = new GQLListSchema('Organization', {
...
+            type: Select,
+            options: ORGANIZATION_TYPES,
+            dataType: 'string',
+            defaultValue: MANAGING_COMPANY_TYPE,
```
Comment (sitozzz) [inline apps/condo/domains/organization/schema/Organization.js:45]: What about all existing organizations? Maybe provide special script that could manage it's type relying on `OrganizationLink` connection?

Comment (SavelevMatthew) [reply apps/condo/domains/organization/schema/Organization.js:45]: See migration

File: apps/condo/domains/organization/schema/OrganizationLink.js (hunk 1 above)
Comment (SavelevMatthew) [reply apps/condo/domains/organization/schema/OrganizationLink.js:59]: I hate them, but for your sake I will add one...

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/organization/schema/Organization.js (hunk 3 above)
Comment (sitozzz) [reply apps/condo/domains/organization/schema/Organization.js:48]: Yup, that's good

**Review by sitozzz — APPROVED** [review]

**Review by Alllex202 — APPROVED** [review]

**Pull request merged**